renderdoc/replay/entry_points.cpp

```diff
@@ -890,6 +1203,208 @@ extern "C" RENDERDOC_API void RENDERDOC_CC RENDERDOC_StartAndroidRemoteServer(co
+  Android::extractDeviceIDAndIndex(host, index, deviceID);
+
+  // Find the path to package
+  string pkgPath = trim(adbExecCommand(deviceID, "shell pm path " + packageName).strStdout);
+  pkgPath.erase(pkgPath.begin(), pkgPath.begin() + strlen("package:"));
+  pkgPath.erase(pkgPath.end() - strlen("base.apk"), pkgPath.end());
+  pkgPath += "lib";
+
+  string layerName = "libVkLayer_GLES_RenderDoc.so";
+
+  // First, see if the application contains the layer
+  if(SearchForAndroidLayer(deviceID, pkgPath, layerName))
+    return true;
+
+  // Next, check a debug location only usable by rooted devices
+  if(SearchForAndroidLayer(deviceID, "/data/local/debug/vulkan", layerName))
+    return true;
+
+  // TODO: Add any future layer locations
+
+  RDCWARN("No RenderDoc layer for Vulkan was found");
+  return false;
+}
+
+string DetermineInstalledABI(const string &deviceID, const string &packageName)
+{
+  RDCLOG("Checking installed ABI for %s", packageName.c_str());
+  string abi;
+
+  string dump = adbExecCommand(deviceID, "shell pm dump " + packageName).strStdout;
+  if(dump.empty())
+    RDCERR("Unable to pm dump %s", packageName.c_str());
+
+  // Walk through the output and look for primaryCpuAbi
+  std::istringstream contents(dump);
+  string line;
+  string prefix("primaryCpuAbi=");
+  while(std::getline(contents, line))
+  {
+    line = trim(line);
+    if(line.compare(0, prefix.size(), prefix) == 0)
+    {
+      // Extract the abi
+      abi = line.substr(line.find_last_of("=") + 1);
+      RDCLOG("primaryCpuAbi found: %s", abi.c_str());
+      break;
+    }
+  }
+
+  if(abi.empty())
+    RDCERR("Unable to determine installed abi for: %s", packageName.c_str());
+
+  return abi;
+}
+
+string FindAndroidLayer(const string &abi, const string &layerName)
+{
+  string layer;
+
+  // Check known paths for RenderDoc layer
+  string exePath;
+  FileIO::GetExecutableFilename(exePath);
+  string exeDir = dirname(FileIO::GetFullPathname(exePath));
+
+  std::vector<std::string> paths;
+
+#if defined(RENDERDOC_LAYER_PATH)
+  string customPath(RENDERDOC_LAYER_PATH);
+  RDCLOG("Custom layer path: %s", customPath.c_str());
+
+  if(FileIO::IsRelativePath(customPath))
+    customPath = exeDir + "/" + customPath;
+
+  // Check to see if layer name was included in custom path
+  if(!endswith(customPath, layerName))
+  {
+    if(customPath.back() != '/')
+      customPath += "/";
+    customPath += layerName;
+  }
+  paths.push_back(customPath);
+#endif
+
+  // clang-format off
+  paths.push_back(exeDir + "/android/libs/" + abi + "/" + layerName);                                   // Windows install
+  paths.push_back(exeDir + "/../share/renderdoc/android/libs/" + abi + "/" + layerName);                // Linux install
+  paths.push_back(exeDir + "/../../build-android/renderdoccmd/libs/" + abi + "/" + layerName);          // Local build
+  paths.push_back(exeDir + "/../../../../../build-android/renderdoccmd/libs/" + abi + "/" + layerName); // macOS build
```

Comment:
Hmm, but if the folder structure is correct in the builds with `lib/` then won't the copy in the temp folder be in an identical structure? I thought you'd just need to run aapt from the appropriate working directory and it would replicate the correct structure. You'd only need the copy if you're changing the directory structure somehow.